group_identifier_assigner.py

```diff
@@ -0,0 +1,266 @@
+    try:
+        return id_type_list.index(id_type)
+    except ValueError:
+        return -1
+
+
+def identifier_matches(id_list, id_type, regex_string):
+    pattern = re.compile(regex_string)
+    index = identifier_index(id_list, id_type)
+    return (index != -1) & (pattern.match(id_list[index]["Identifier"]) is not None)
+
+
+def add_identifier_to_group(gid, type, identifier_name):
+    new_identifier_info = {
+        "Version": "1.0",
+        "Type": type,
+        "Identifier": identifier_name,
+        "Login": False,
+        "Person": {"Type": "Group", "Id": str(gid)},
+        "Status": "Active",
+    }
+    data = {
+        "RequestType": "Identifiers",
+        "Version": "1.0",
+        "Identifiers": [new_identifier_info],
+    }
+    return call_api3(POST, "identifiers.json", data)
+
+
+def parse_options(args):
+    try:
+        ops, args = getopt.getopt(args, "u:c:d:f:e:o:t:T:h")
+    except getopt.GetoptError:
+        usage()
+
+    if args:
+        usage("Extra arguments: %s" % repr(args))
+
+    passfd = None
+    passfile = None
+
+    for op, arg in ops:
+        if op == "-h":
+            usage()
+        if op == "-u":
+            options.user = arg
+        if op == "-c":
+            options.osg_co_id = int(arg)
+        if op == "-d":
+            passfd = int(arg)
+        if op == "-f":
+            passfile = arg
+        if op == "-e":
+            options.endpoint = arg
+        if op == "-o":
+            options.outfile = arg
+        if op == "-t":
+            options.min_timeout = float(arg)
+        if op == "-T":
+            options.max_timeout = float(arg)
+
+    user, passwd = getpw(options.user, passfd, passfile)
+    options.authstr = mkauthstr(user, passwd)
+
+
+def main(args):
+    parse_options(args)
+
+    # get groups with 'OSPool project name' matching "Yes-*" that don't have a 'OSG GID'
+
+    co_groups = get_osg_co_groups()["CoGroups"]
+    highest_osggid = 0
+    projects_to_assign_identifiers = []
+
+    for group in co_groups:
+        gid = group["Id"]
+        identifier_data = get_co_group_identifiers(gid)
+
+        if identifier_data:
+            identifier_list = identifier_data["Identifiers"]
+
+            project_id_index = identifier_index(identifier_list, "ospoolproject")
+            if project_id_index != -1:
+                project_id = str(identifier_list[project_id_index]["Identifier"])
+                is_project = re.compile(OSPOOL_PROJECT_PREFIX_STR + "*").match(project_id) is not None
+            else:
+                is_project = False
+
+            osggid_index = identifier_index(identifier_list, "osggid")
+            if osggid_index != -1:
+                highest_osggid = max(highest_osggid, int(identifier_list[osggid_index]["Identifier"]))
+            elif is_project is True:
+                project_name = project_id.replace(OSPOOL_PROJECT_PREFIX_STR, "", 1).lower()
+                projects_to_assign_identifiers.append(tuple([gid, project_name]))
```

Comment:
You can tack on an extra comma to let Python know you want a 2-tuple using parentheses (i.e. without converting a list):
```suggestion
                projects_to_assign_identifiers.append((gid, project_name,))
```